app/src/components/common/Table.vue

```diff
@@ -44,21 +61,90 @@
 </template>
 
 <script lang="ts">
-import { TableColumn } from '@/types'
+import { TableColumn, SortOrder, SortRule } from '@/types'
 import { Vue, Component, Prop } from 'vue-property-decorator'
 
-@Component
+@Component({
+  computed: {
+    SortOrder () {
+      return SortOrder
+    }
+  }
+})
 export default class Table extends Vue {
   @Prop({ required: true }) columns!: TableColumn[]
   @Prop({ required: true }) data!: Array<{ [key: string]: string }>
+  sortedData: Array<any> = []
+  sortingColumn: TableColumn | undefined = undefined
+  sortingRules : SortRule[] = [
+    {
+      currSort: SortOrder.NEUTRAL,
+      nextSort: SortOrder.ASC
+    },
+    {
+      currSort: SortOrder.ASC,
+      nextSort: SortOrder.DESC,
+      compareFn: (a: any, b: any) =>
+        typeof a === 'string' && typeof b === 'string'
+          ? a.toLowerCase().localeCompare(b.toLowerCase())
+          : a > b ? 1 : -1
+    },
+    {
+      currSort: SortOrder.DESC,
+      nextSort: SortOrder.NEUTRAL,
+      compareFn: (a: any, b: any) =>
+        typeof a === 'string' && typeof b === 'string'
+          ? b.toLowerCase().localeCompare(a.toLowerCase())
+          : a > b ? -1 : 1
+    }
+  ]
+
+  mounted () {
+    this.columns.forEach((column) => {
+      if (column.isSortable && !column.sortOrder) {
+        this.$set(column, 'sortOrder', SortOrder.NEUTRAL)
+      }
+    })
+  }
 
   getLabel (column: TableColumn): string {
-    return column.label ? column.label : ''
+    return column.label || ''
   }
 
   getEntry (column: TableColumn, item: { [key: string]: string }): string {
     return item[ column.key ]
   }
+
+  setSortingColumn (column: TableColumn) {
+    if (column.isSortable) {
+      this.sortingColumn = column
+      column.sortOrder = this.sortingRules.find(
+        (rule) => rule.currSort === column.sortOrder
+      )!.nextSort
+      this.updateSortedData()
+    }
+  }
+
+  get displayData () : Array<any> {
+    this.sortedData = this.data.slice()
+    this.updateSortedData()
+    return this.sortedData
+  }
+
+  updateSortedData () {
+    if (this.sortingColumn) {
```

Comment:
In what cases `sortingColum` does not exist? Maybe it will be clearer if we check if the column `isSortable`?

Reply:
Initial sortingColum is undefined, so that is the reason why I check if it exists. updateSortedData can be called if the data passed in props is changed and will be automatically sorted, however, if we haven't set any sorting column yet and data changes, there could be an error of accessing an undefined variable. I check if the column is sortable in setSortingColumn function when we press on the table header.